payroll/views.py

```diff
@@ -298,7 +302,9 @@ def get_queryset(self):
 
 def flush_cache(request, secret_key):
     if secret_key == CACHE_SECRET_KEY:
-        cache.clear()
+        for cache_label in settings.CACHES.keys():
```

Comment:
the docs say that caches is dict-like, so can you do 

```python
for cache in caches.values():
    cache.clear()
```

if so that would be clearer intent to clear all caches.

Reply:
I did this initially, but alas it is not _that_ dict-like. 

```
AttributeError: 'CacheHandler' object has no attribute 'values'`
```

It has an `all()` method, but it doesn't return anything, even with the cache configured.

```
(Pdb) caches
<django.core.cache.CacheHandler object at 0x7f41b26248d0>
(Pdb) dir(caches)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_caches', 'all']
(Pdb) caches.all()
dict_values([])
```

Reply:
I'm actually very confused by [the `caches` implementation](https://github.com/django/django/blob/c494d90c195a739a7298b073eaa6ed987c2fd0bc/django/core/cache/__init__.py#L57-L84). It uses an instance of threading `local()` for caches... which doesn't seem to contain the caches until they are accessed:

```
(Pdb) caches
<django.core.cache.CacheHandler object at 0x7fb9a075f8d0>
(Pdb) caches._caches
<_thread._local object at 0x7fb9a0771150>
(Pdb) caches._caches.caches
*** AttributeError: '_thread._local' object has no attribute 'caches'
(Pdb) caches['default']
<django.core.cache.backends.db.DatabaseCache object at 0x7fb99eaac630>
(Pdb) caches._caches.caches
{'default': <django.core.cache.backends.db.DatabaseCache object at 0x7fb99eaac630>}
(Pdb) caches['vary_on_setting']
<django.core.cache.backends.db.DatabaseCache object at 0x7fb990d7b4e0>
(Pdb) caches._caches.caches
{'vary_on_setting': <django.core.cache.backends.db.DatabaseCache object at 0x7fb990d7b4e0>, 'default': <django.core.cache.backends.db.DatabaseCache object at 0x7fb99eaac630>}
```

But the practical effect is that `caches.all()` is empty in the flush view.

Reply:
what a pain. thanks for looking into it.